innerprod/fullysec/damgard_multi.go

```diff
@@ -29,169 +31,209 @@ import (
 // Abdalla, Catalano, Fiore, Gay, and Ursu:
 // "Multi-Input Functional Encryption for Inner Products:
 // Function-Hiding Realizations and Constructions without Pairings".
+// The participants in the scheme are clients and a central authority.
+// The central authority generates keys for each client so that client i
+// encrypts vector x_i. The scheme allows the central authority to
+// generate a key_Y, depending on a matrix Y with rows y_i, so that
+// given key_y and the ciphertext the decryptor can compute value
+// Σ_i <x_i, y_i> (sum of dot products).
+
+// DamgardMultiClient is a struct in DamgardMulti scheme, that holds
+// all the shared parameters, and can represent the central authority
+// or the decryptor.
 type DamgardMulti struct {
-	// number of encryptors
-	Slots int
+	// number of clients
+	NumClients int
+	Bound      *big.Int
+	*Damgard
+}
+
+// DamgardMultiClient represents a single client for the DamgardMulti scheme.
+type DamgardMultiClient struct {
+	Bound *big.Int
```

Comment:
Isn't the bound included in `Damgard.Params`?

Reply:
Good point, but there is a difficulty with that. In multi-client scheme encryption is done by encrypting x_i + o, where o is a one time pad. Thus the value that is encrypted can be bigger then the bound and `Damgard` scheme would reject it.

Reply:
OK. Then just add a couple of comments explaining this.